### PR TITLE
register available_managed_shm metric only for readout-proxy

### PR DESCRIPTION
**Pull request by ehellbar**
this provides a quick fix for https://its.cern.ch/jira/browse/O2-4234

now there is the comment about the ugliness of the condition in this place. @ktf to you have a better general solution for this in mind? At the moment, I can only think of more ugly ways to decouple the sending of the other metrics from the `available_managed_shm` for the `readout-proxy` only...

### CODE REVIEW

**Review by davidrohr — APPROVED** [review]
Looks good to me, if @ktf could confirm that it will work.
I guess there is no good way to test it?
Or perhaps you can test with the stdout backend of the metrics, and then just check on the console that you get it only from readout-proxy?

File: Framework/Core/src/CommonServices.cxx [hunk 1]
```diff
@@ -1105,6 +1108,9 @@ o2::framework::ServiceSpec CommonServices::dataProcessingStats()
                    .sendInitialValue = true}};
 
       for (auto& metric : metrics) {
+        if (metric.metricId == (int)ProcessingStatsId::AVAILABLE_MANAGED_SHM_BASE + (runningWorkflow.shmSegmentId % 512) && spec.name.compare("readout-proxy") != 0) {
```
Comment (ktf) [inline Framework/Core/src/CommonServices.cxx:1111]: Is "readout-proxy" the only name we have?

**Comment by ktf** [comment]
Probably the correct way of doing this would be to have the metric specification as part of a service and allow devices to add their own customisations... I would say good enough for now (modulo the comment on the name) and we can have a better fix later.

File: Framework/Core/src/CommonServices.cxx (hunk 1 above)
Comment (ehellbar) [reply Framework/Core/src/CommonServices.cxx:1111]: it is the only name we use in the online processing, and we only display it for this task in Grafana. The calibration workflows have different input proxy names each, and (at least so far) we don't look at their shm.

**Comment by ehellbar** [comment]
So I tested it locally with the metrics display in the GUI, using a simple workflow and replacing the check for `readout-proxy` with another task name.

I also used the monitoring stdout backend with the o2-dpl-raw-proxy, using two different names:
```
o2-dpl-raw-proxy  --proxy-name readout-proxy --driver-client-backend "stdout://" -b > output_proxy-name_readout-proxy.log
grep -nr '\[METRIC\]' output_proxy-name_readout-proxy.log  | wc -l
      31
grep -nr '\[METRIC\] available_managed_shm' output_proxy-name_readout-proxy.log  | wc -l
       1
```

```
o2-dpl-raw-proxy  --proxy-name dummy-proxy --driver-client-backend "stdout://" -b > output_proxy-name_dummy-proxy.log
grep -nr '\[METRIC\]' output_proxy-name_dummy-proxy.log  | wc -l
      30
grep -nr '\[METRIC\] available_managed_shm' output_proxy-name_dummy-proxy.log  | wc -l
       0
```

**Pull request merged**